lib/record.rb

```diff
@@ -0,0 +1,20 @@
+class Record
+  attr_reader :amount, :date, :type
+
+  def initialize(amount, date)
+    @amount = check_validity(amount)
+    @date = date
+    @type = set_type
+  end
+
+  private
+
+  def check_validity(amount)
```

Comment:
It feels like `check_validity` has two responsibilities - throwing if the amount is invalid, and also returning the amount.  Could these be separated?